app/models/solidus_paypal_braintree/gateway.rb

```diff
@@ -292,7 +301,9 @@ def transaction_options(source, options, submit_for_settlement = false)
       end
 
       params[:channel] = "Solidus"
-      params[:options] = { store_in_vault_on_success: true }
+      params[:options] = {
+        store_in_vault_on_success: (store_in_vault && preferred_paypal_flow == 'vault')
```

Comment:
Could we move this `preferred_paypal_flow == 'vault' ` into the `store_in_vault` method? I think it makes more sense in there.

Reply:
@tvdeyen the problem is that `store_in_vault_on_success: true` will fail if the flow is not `vault`, so I wanted to avoid potential issues with that.

Reply:
@tvdeyen fixed.